### PR TITLE
Implemented: stock schema design, action to fetch stock for multiple products and defined tranformation rule for stock response(#2v1pmpv)

### LINKED ISSUE #36: Define schema for stock
## What are the acceptance criteria?

- [x] Function to get stock for product
- [x] Stock Schema
- [x] Transformation rule

## Can you complete this feature request by yourself?

- [x] YES
- [ ] NO

### PR DESCRIPTION
**Pull request by ymaheshwari1**
### Related Issues

Closes #36 

### Short Description and Why It's Useful

- Defined an action to call checkInventory api which fetches the stock for multiple product at once
- Added tranformation rule to return the response from oms package as per the stock schema design

### Contribution and Currently Important Rules Acceptance

- [x] I read and followed [contribution rules](https://github.com/hotwax/oms-api/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: src/modules/stock/index.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import api from "@/api";
+import { hasError } from "@/util";
+import { Stock, Response } from "@/types"
+import { transform } from "node-json-transform";
+import { stockTransformRule } from "@/mappings/stock";
+
+async function fetchProductsStock(productIds: Array<string>, facilityId?: string): Promise<Array<Stock> | Response> {
```
Comment (adityasharma7) [inline src/modules/stock/index.ts:7]: As we are not fetching overall inventory in this method, we should rename based upon use case

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/modules/stock/index.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import api from "@/api";
+import { hasError } from "@/util";
+import { Stock, Response } from "@/types"
+import { transform } from "node-json-transform";
+import { stockTransformRule } from "@/mappings/stock";
+
+async function fetchProductsStockAsPerFacility(productIds: Array<string>, facilityId?: string): Promise<Array<Stock> | Response> {
```
Comment (adityasharma7) [inline src/modules/stock/index.ts:7]:
```suggestion
async function fetchProductsStockAtFacility(productIds: Array<string>, facilityId?: string): Promise<Array<Stock> | Response> {
```

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/modules/stock/index.ts (hunk 2 above)
Comment (ymaheshwari1) [reply src/modules/stock/index.ts]: Improved method name

File: src/modules/stock/index.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  // but as we will always fetch data for the fetched records which will be as per the viewSize
+  // assuming that the value will never be 100 to show
+
+  // When facility is not being passed stock for all the facilities will be fetched
+  const query = {
+    "filters": {
+      "productId": productIds,
+      "productId_op": "in"
+    },
+    "viewSize": productIds.length,
```
[the remark this answers is not in the document]
Comment (ymaheshwari1) [reply src/modules/stock/index.ts]: Improved calculation of viewSize and added some comments

**Pull request merged**